Sources/Core/Utilities/ExifManager/ExifManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  ExifManager.swift
+//  PovioKit
+//
+//  Created by Marko Mijatovic on 16/02/2023.
+//  Copyright © 2023 Povio Inc. All rights reserved.
+//
+
+import Foundation
+import ImageIO
+import PovioKitPromise
+
+/// A wrapper around Image I/O framework APIs that can be used to modify EXIF and other image metadata without recompressing the image data.
+public final class ExifManager {
+  private let source: ExifImageSource
+  
+  public init(source: ExifImageSource) {
+    self.source = source
+  }
+}
+
+// MARK: - Public Methods
+public extension ExifManager {
+  /// Read EXIF metadata from the provided image source
+  /// - Returns: Dictionary with EXIF values as a Promise or ``ExifError``
+  func readExif() -> Promise<[String: Any]> {
+    guard let imageSource = getImageSource() else {
+      return .error(ExifError.createImageSource)
+    }
+    guard let imageProperties = CGImageSourceCopyPropertiesAtIndex(imageSource, 0, nil) as? [String: Any] else {
+      return .error(ExifError.getImageProperties)
+    }
+    return .init(fulfill: imageProperties)
+  }
+    
+  /// Update EXIF metadata with the new values
+  ///
+  /// Keys for the new values must be part of the [EXIF Dictionary Keys](https://developer.apple.com/documentation/imageio/exif_dictionary_keys)
+  /// - Parameter newValue: Dictionary with the new EXIF values
+  /// - Returns: New image Data as a Promise or ``ExifError``
+  func updateExif(_ newValue: [CFString: String]) -> Promise<Data> {
```

Comment:
If we rename the manager, maybe we can rename this method as well to `update` 🤷‍♂️